### PR TITLE
Fix cagra multi CTA bug

### PR DESCRIPTION
**Pull request by enp1s0**
rel: #1575

The main causes of the bug are the wrong memory allocation size and the wrong number of threads to be launched.

### CODE REVIEW

**Comment by enp1s0** [comment]
@tfeher Can you review the code?

**Review by cjnolet — APPROVED** [review]
LGTM, thanks @enp1s0! Once conflicts are resolved, I can merge this.

**Pull request merged**
